### ISSUE TITLE
Target suffix globs are optionally matched

### ISSUE THREAD

**Issue by jayvdb** [EFForg/https-everywhere#18892]
[//]: # (Thank you for reporting an issue to HTTPS Everywhere.)
[//]: # (We welcome input from users on improving this project.)
[//]: #
[//]: # (Please help us by following this issue template.)
[//]: # (You can delete all blank lines and all lines starting)
[//]: # (with the comment marker, such as this one.)

Type: feature request

[//]: # (Include any other relevant information below. Thank you again for)
[//]: # (helping to improve HTTPS Everywhere.)

CONTRIBUTING.md discourages suffix globs on targets, and shows the Google rules as examples of where it is appropriate.

I was a bit surprised to find that `Google.tld_Subdomains.xml` **only** includes

>  <target host="accounts.google.com.*" />

and that suffix glob is supposed to match the test case `http://accounts.google.com/`

The same occurs in `Google.xml`, with target `www.google.com.*` and test cases `http://www.google.com/about`.

This non-mandatory matching of suffix globs is only needed by tests for those two Google rulesets, but there are a few other rulesets using target suffix `.*` (only ` HSBC.xml` and `Ticketmaster.xml` it seems) so it is possibly being utilised in those other rulesets without tests ensuring it works.

However this is messy to implement, and IMO should be replaced with explicit targets without the `.*` when that is needed.  Foremost, this would make it obvious that extra test cases are needed for those extra targets.  Also I am pretty sure that any library implementing `.*` target suffixes will assume they are mandatory, and so these very important targets will not work.  The libraries I have looked at which reside in other repos are not running the .xml test suite, and often have quite small tests, so I would wager this is a bug in most implementations.

In addition, it means extra effort is needed to implement sanely/safely, as for e.g. `hsbc.*` in `HSBC.xml` surely should not also match `hsbc` (possibly a hostname on local domain?), and `hsbc.co.*` shouldnt match `hsbc.co` (doesnt exist, but could be registered like `google.co`, and could be legitimately owned by someone other than the owner of `hsbc.co.uk` and friends, depending on the policies of the .co domain, owned by U.S. for-profit Neustar).  Even if those cases are not highly probable/problematic, it is still extra lookups needed to resolve the appropriate ruleset to use.

**Comment by cschanaj** [comment]
IIRC, `<target host="accounts.google.com.*" />` will not match `accounts.google.com`. You might want to check [background-scripts/util.js#L100](https://github.com/EFForg/https-everywhere/blob/master/chromium/background-scripts/util.js#L100) for the current implementation. 

We do not include preload domains in the rulesets, that's might by why you see only a relatively small number of subdomain included in Google's rulesets. Feel free to correct me if I missed anything.

**Comment by pipboy96** [comment]
@cschanaj Correct, if domain is preloaded in current versions of Chrome, Firefox and Tor Browser, it is excluded.

**Comment by jayvdb** [comment]
Ok, great to know this is a ruleset bug as opposed to a non-intuitive design/implementation.

**Comment by pipboy96** [comment]
@jayvdb This is both a ruleset bug (relying on this behavior instead of adding explicit targets) and code bug (this behavior is extremely non-intuitive and may be completely unintended).
